docs/cri/config.md

```diff
@@ -578,6 +579,10 @@ version = 2
     # bin_dir is the directory in which the binaries for the plugin is kept.
```

Comment:
The same note about only one of bin_dir or bin_dirs should be used should be on this as well. I half wonder what would be more sane, this new field solely being additional directories to check on top of bin_dir, or this mutually exclusive model. I don't have strong opinions

Reply:
> The same note about only one of bin_dir or bin_dirs should be supplied should be on this comment as well

Updated.

> this new field solely being additional directories to check on top of bin_dir, or this mutually exclusive model

I feel the `mutually exclusive model` is simpler and straightfoward. And long term I think we should deprecate `bin_dir` and only keep `bin_dirs`(?) Then we only need to change the `CniConfig` struct (and `ValidateRuntimeConfig`) since we already moved all `bin_dir` usage to `bin_dirs` in our code.

Reply:
I'd love to deprecate bin_dir, I wish we would've gotten to it before 2.0.. This is fine with me

Reply:
let's say we allow using both:
if the bin_dir is in bin_dirs[0] it shouldn't matter just don't combine..
if the bin_dir is in a later bin_dirs[] and it's not a dup then order matters and we don't know which to use (validation error) 
if the bin_dir is not in bin_dirs we're either broken or missing a directory if we ignore bin_dir
if the bin_dir is empty we don't care.. and check for any in bin_dirs[]

let's say we disallow using both:
deprecate and migrate bin_dir to bin_dirs 
add a validation fail if both are used simultaneously..

I vote disallow both.. allowing both is to complicated due to ordering imo.

Reply:
> let's say we allow using both:
> ...

Exactly (and thanks for the detailed examples) :) it both complicates our logic and may hide user config issues.

> I vote disallow both.. allowing both is to complicated due to ordering imo.

Agreed. This is the direction this PR aims to.

> deprecate and migrate bin_dir to bin_dirs

Yeah I think we should deprecate `bin_dir`. I wonder if we should just add a `Deprecated: ` comment above `NetworkPluginBinDir` directly in this PR (which I prefer), or we should do it later? 🤔 (still trying to get familiar with how we deprecate things :))

Reply:
@samuelkarp is our deprecation expert we can do the migration in a follow up later in the year..